src/devices/Ssd13xx/Ssd1309.cs

```diff
@@ -0,0 +1,151 @@
+
+using System;
+using System.Device.Gpio;
+using System.Device.Spi;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Iot.Device.Ssd13xx.Commands;
+using Ssd1306Cmnds = Iot.Device.Ssd13xx.Commands.Ssd1306Commands;
+
+namespace Iot.Device.Ssd13xx
+{
+    /// <summary>
+    /// A single-chip CMOS OLED/PLED driver with controller for organic/polymer
+    /// light emitting diode dot-matrix graphic display system.
+    /// </summary>
+    public class Ssd1309 : Ssd13xx
+    {
+        private readonly int _dc;
+        private readonly int _rst;
+
+        private GpioController? _gpioController;
+
+        /// <summary>
+        /// Initializes new instance of Ssd13069 device that will communicate using SPI
+        /// in a non-traditional "4-wire SPI" mode that uses DC and RST GPIO pins to switch
+        /// between Data/Command instructions and reset behavior.
+        /// A single-chip CMOS OLED/PLED driver with controller for organic/polymer
+        /// light emitting diode dot-matrix graphic display system.
+        /// </summary>
+        /// <param name="spiDevice">The SPI device used for communication.</param>
+        /// <param name="width">Width of the display. Typically 128 pixels</param>
+        /// <param name="height">Height of the display, variants with 32 or 64 rows exist</param>
+        /// <param name="gpioController">GPIO controller supplied by Board.CreateGpioController()</param>
+        /// <param name="dcPin">GPIO pin for Data/Command switch</param>
+        /// <param name="rstPin">GPIO pin for Reset behavior</param>
+        public Ssd1309(SpiDevice spiDevice, GpioController gpioController, int dcPin, int rstPin, int width, int height)
+        : base(spiDevice, width, height)
+        {
+            _dc = dcPin;
+            _rst = rstPin;
+            _gpioController = gpioController;
+            _gpioController.OpenPin(_dc, PinMode.Output, PinValue.Low);
+            _gpioController.OpenPin(_rst, PinMode.Output, PinValue.High);
+            Reset();
+            Initialize();
+        }
+
+        /// <summary>
+        /// Sends command to the device
+        /// </summary>
+        /// <param name="command">Command being sent</param>
+        public void SendCommand(ISsd1309Command command) => SendCommand((ICommand)command);
+
+        /// <summary>
+        /// Sends command to the device
+        /// </summary>
+        /// <param name="command">Command being sent</param>
+        public override void SendCommand(ISharedCommand command) => SendCommand(command);
+
+        /// <summary>
+        /// Send a command to the display controller.
+        /// </summary>
+        /// <param name="command">The command to send to the display controller.</param>
+        private void SendCommand(ICommand command)
+        {
+            Span<byte> commandBytes = command?.GetBytes();
+
+            _gpioController?.Write(_dc, PinValue.Low);
```

Comment:
The main difference between this and the existing I2C implementation is that the DC pin is set to LOW for commands and HIGH for data. the SSD1306 with I2C uses a control byte here.